### DISCUSSION TITLE
[Berry] Possible to set MQTT details?

### DISCUSSION THREAD

**Discussion by Slyke** [arendst/Tasmota#15368] [Q&A] [answered]
Hello,

Is it possible to set MQTT details, such as host and port with a Berry script? Want to load some configs from a file on an SD card and Tasmota. From my current understanding Tasmota requires a reboot before the new settings are applied?

I did find this: https://tasmota.github.io/docs/Berry/#mqtt-module but it's just for publishing data.

**Comment by sfromis** [comment, marked as answer]
In general, you can use all Tasmota command from Berry too, like `tasmota.cmd('MqttHost my.broker')`
Yes, reboot is needed and should happen automatically when having executed the command. If you need more than one setting updated, you may consider using a `Backlog` command to put them all into one string.

**Reply by Slyke** [reply]
Ahh ok. I was going to place this into the `autoexec.be` file, but that'd cause a boot loop. Looks like it'd need to be manually triggered, not a big deal in any case.

**Reply by sfromis** [reply]
Well, if you check the values before making a change, you can still do it "hidden".

**Reply by Slyke** [reply]
Yep, that'd work. Just worried that if it doesn't save or fails to correctly compare current settings with saved to SDCard that it'll get stuck in a boot loop. Could make it watch for a GPIO to go high and then disable checking, but extra complexity.
